File: TPs/TP1/TP01Q12.c
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int main()
{
	char palavra[100];
	fgets(palavra,sizeof(palavra), stdin);
	palavra[strcspn(palavra, "\n")] = '\0';
	int tamanho = strlen(palavra);

	bool maiuscula;
	bool minuscula;
	bool especial;
	bool numero;
	bool check1 = false;
	bool check2 = false;
	bool check3 = false;
	bool check4 = false;

	for(int i =0 ; i < tamanho; i++) {

		if(palavra[i] < 'A' || palavra[i] > 'Z' && check1 == false  ) {

			maiuscula = true;
			check1 = true;

		} else if(palavra[i] < 'a' || palavra[i] > 'z' && check2 == false ) {

			minuscula = true;
			check2 = true;
		} else if(ispunct(palavra[i]) && check3 == false ) {

			especial = true;
			check3 = true;

		} else if(palavra[i] >= '0' && palavra[i] <= '9' && check4 == false ) {

			numero = true;
			check4 = true;
		}


	}

	if(check1) {
		printf("SIM para maiuscula \n ");
	} else {
		printf("NCO para maiuscula \n");
	}

	if(check2) {
		printf("SIM para minuscula \n ");
	} else {
		printf("NCO para minuscula \n");
	}

	if(check3) {
		printf("SIM para especial \n");
	} else {
		printf("NCO para especial \n");
	}


	if(check4) {
		printf("SIM para numero  \n");
	} else {
		printf("NCO para numero \n");
	}


	/*
	if(check1 && check2 && check3 && check4){
	   printf("SIM");
	}else{
	   printf("NCO");
	}

	*/



	return 0;
}

File: TPs/TP1/TP01Q12.java
public class TP01Q12
{
	public static void main(String[] args) {

		boolean isfim = false;
		do {

			String palavra = MyIO.readString();
			char [] senha = palavra.toCharArray();
			int tamanho = senha.length;
			boolean maiuscula = false;
			boolean minuscula = false;
			boolean especial = false;
			boolean numero = false;
			boolean check1 = false;
			boolean check2 = false;
			boolean check3 = false;
			boolean check4 = false;
			int i = 0;
			
			if(palavra.equalsIgnoreCase("FIM")){
			    isfim = true;
			    break;
			}

			if(tamanho < 8) {
				i = tamanho;
			}


			for( i = 0 ;  i < tamanho; i++) {

				if(Character.isUpperCase(senha[i]) && check1 == false) { // isupper verifica se o vetor e Maiusculo

					maiuscula = true;
					check1 = true;

				} else if(Character.isLowerCase(senha[i]) && check2 == false ) { // islower verifica se o vetor C) minusculo

					minuscula = true;
					check2 = true;

				} else if(Character.isLetterOrDigit(senha[i]) && check3 == false ) { // ispunct verifica se o vetor C) especial

					especial = true;
					check3 = true;

				} else if(Character.isDigit(senha[i]) && check4 == false ) { // isdigit verifica se o vetor C) numero

					numero = true;
					check4 = true;
				}

			}


			if(check1 && check2 && check3 && check4) {
				MyIO.println("SIM");
			} else {
				MyIO.println("NAO");
			}

		} while(isfim != true);

	}
}



